content/posts/2021-01-26-ggplot-regression-line.it.Rmarkdown

```diff
@@ -0,0 +1,80 @@
+
+Di defaut, `geom_smooth()` aggiunge una curva LOESS ai dati. Tuttavia non è quello che vogliamo. Per far sì che `geom_smooth()` disegni una retta di regressione lineare dobbiamo impostare il parametro `method` su `"lm"`, che è l'abbreviazione di "linear model" (modello lineare).
+
+```{r ggplot2_scatter_plot_with_linear_regression_line, message=FALSE}
+p + geom_smooth(method = "lm")
+```
+
+<!-- The gray shading around the line represents the 95% confidence interval. You can change the confidence interval level by changing the `level` parameter. A value of `0.8` represents a 80% confidence interval. -->
+
+La parte in grigio chiaro attorno alla linea indica l'intervallo di confidenza al 95%. Puoi cambiare il livello dell'intervallo di confidenza modificando il parametro `level`. Un valore di `0.8` corrisponde a un intervallo di confidenza all'80%.
+
+```{r ggplot2_linear_regression_line_confidence_interval, message=FALSE}
+p + geom_smooth(method = "lm", level = 0.8)
+```
+
+<!--If you don't want to show the confidence interval band at all, set the `se` parameter to `FALSE`. -->
+
+Se non vuoi inserire l'intervallo di confidenza nel grafico, imposta il parametro `se` come `FALSE`.
+
+```{r ggplot2_linear_regression_line_without_confidence_interval, message=FALSE}
+p + geom_smooth(method = "lm", se = FALSE)
+```
+
+<!-- Sometimes a line is not a good fit to the data but a polynomial would be. So, how to add a polynomial regression line to a plot? To do so, we will still have to use `geom_smooth()` with `method = "lm"` but in addition specify the `formula` parameter. By default, `formula` is set to `y ~ x` (read: `y` as a function of `x`). To draw a polynomial of degree `n` you have to change the formula to `y ~ poly(x, n)`. Here's an example fitting a 2nd degree (quadratic) polynomial regression line. -->
+
+Qualche volta una retta potrebbe non essere la soluzione più adatta per i dati, mentre potrebbe esserlo una polinomiale. Come possiamo aggiungere una linea di regressione polinomiale ad un grafico? Per farlo dobbiamo comunque usare `geom_smooth()` con `method = "lm"` ma in aggiunta dobbiamo specificare il parametro `formula`. Di default, `formula` è impostato su `y ~ x` (si legge `y` in funzione di `x`). Per tracciare una polinomiale di grado `n` bisogna cambiare la formula in `y ~ poly(x, n)`. Ecco un esempio che utilizza una curva di regressione polinomiale di secondo grado (quadratica).
+
+```{r ggplot2_polynomial_regression_line}
+ggplot(mtcars, aes(qsec, hp)) +
+  geom_point() +
+  geom_smooth(method = "lm", formula = y ~ poly(x, 2))
+```
+
+<!-- Now it's your turn! Start a new R session, load some data, and create a ggplot with a linear regression line. Happy programming! -->
+
+Ora è il tuo turno! Apri una nuova sessione in R, carica dei dati e crea un ggplot con una retta di regressione lineare. Buona programmazione!
```

Comment:
Please add a sentence like "This post has been kindly translated from the original English by <your name or twitter handle with a link to your twitter/website/GitHub (whatever you like)>".

Reply:
Ok i've added it and pushed the commit. Do I have to update the PR somehow?

Reply:
No, it's automatically added to the PR 🙂